### PR TITLE
Add GetPathToRoot method for markdown object

### PR DESCRIPTION
**Pull request by yufeih**
[AB#278675](https://dev.azure.com/ceapex/Engineering/_workitems/edit/278675/)

https://github.com/lunet-io/markdig/issues/461
https://github.com/lunet-io/markdig/issues/460

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6360)

### CODE REVIEW

File: src/docfx/lib/markdown/EnsureParentExtension.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                case Block block when block.Parent is null:
+                    block.SetData(s_parentKey, parent);
+                    break;
+
+                case Inline inline when inline.Parent is null:
+                    inline.SetData(s_parentKey, parent);
+                    break;
+            }
+
+            switch (obj)
```
Comment (live1206) [inline src/docfx/lib/markdown/EnsureParentExtension.cs:61]: combine these two `switch`?

File: src/docfx/lib/markdown/EnsureParentExtension.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                obj = parent;
+            }
+        }
+
+        private static void EnsureParent(MarkdownObject obj, MarkdownObject? parent)
+        {
+            switch (obj)
+            {
+                case Block block when block.Parent is null:
+                    block.SetData(s_parentKey, parent);
```
Comment (live1206) [inline src/docfx/lib/markdown/EnsureParentExtension.cs:53]: what if the input `parent` is `null`?

File: src/docfx/lib/markdown/EnsureParentExtension.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+namespace Microsoft.Docs.Build
+{
+    /// <summary>
+    /// Allow trace back to root node for any markdown object: https://github.com/lunet-io/markdig/issues/461
+    /// </summary>
+    internal static class EnsureParentExtension
+    {
+        private static readonly object s_parentKey = new object();
+
+        public static MarkdownPipelineBuilder UseEnsureParent(this MarkdownPipelineBuilder builder)
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/EnsureParentExtension.cs:19]: can we add some tests on this extension?

Comment (yufeih) [reply src/docfx/lib/markdown/EnsureParentExtension.cs:19]: The effect of this extension `GetPathToRoot` used by `GetFilePath` in this PR has been exercised thoroughly. (see the 100% code coverage on `EnsureParentExtension.cs`). We generally don't write UT in docfx on specific methods, especially when it ties to some framework.

**Pull request merged**